### PR TITLE
DAOS-1129 rebuild: Rename field in daos_rebuild_status struct

### PR DESCRIPTION
**Pull request by chrisphoffman**
*Renamed rs_done field to rs_running in daos_rebuild_status struct. This was due to the rebuild could be "done" without ever running.

### CODE REVIEW

File: src/include/daos_pool.h [hunk 1]
```diff
@@ -96,23 +96,25 @@ struct daos_rebuild_status {
...
 	 */
 	uint64_t		rs_toberb_obj_nr;
-	/** # rebuilt objects, it's non-zero only if rs_done is 1 */
+	/** # rebuilt objects, it's non-zero only if rs_running is 1 */
 	uint64_t		rs_obj_nr;
-	/** # rebuilt records, it's non-zero only if rs_done is 1 */
+	/** # rebuilt records, it's non-zero only if rs_running is 1 */
 	uint64_t		rs_rec_nr;
+	/** # progress of indicator for rebuild in percentage */
+	int32_t			rs_progress;
```
Comment (dmiter) [inline src/include/daos_pool.h:117]: Where this is used? I don't see it in this patch.

Comment (chrisphoffman) [reply src/include/daos_pool.h:117]: It's currently not used, @skirvan44 requested this be added to the struct. He can provide more input.

**Review by bfaccini — APPROVED** [review]

**Review by dmiter — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by jolivier23 — CHANGES_REQUESTED** [review]
Looks like this needs a merge of latest master branch

**Review by ashleypittman — COMMENTED** [review]
This is an API change which we need to be cautious of.

The field is also a uint32_t but could easily be a bool instead if we do want to change it though.

**Review by bfaccini — APPROVED** [review]

**Comment by bfaccini** [comment]
As I believe it will appear in the final/landed commit msg, may be you could manually modify the original message for this PR to remove the "*Added a progress field to daos_rebuild_status struct" comment which no longer applies.

**Comment by jolivier23** [comment]
> This is an API change which we need to be cautious of.
> 
> The field is also a uint32_t but could easily be a bool instead if we do want to change it though.

As far as I can tell, it's not really an API change.  It's just changing the name of the field but that field is the same size and the name of the field isn't part of the ABI.

**Review by jolivier23 — CHANGES_REQUESTED** [review]
I don't think this is quite what the ticket wanted to do.  We can't just rename the field because rs_done != rs_running.   To change it to rs_running, we would need to change the implementation to ensure that it's set if rebuild is running.   In the spirit of the ticket, just renaming rs_done to rs_inactive would fix the stated concern that rs_done implies it did something whereas rs_inactive just means it's not running.   To determine if it has run, one would need to check the status of the pool to see if there are DOWNOUT targets though that may be insufficient.

It may be worth asking whether the intention of the ticket was a feature request to provide a "running" flag or just pointing out that "done" is misleading or perhaps both.   Anyway, if you are going to fix the meaning of the flag, I think just replacing rs_running in your patch with rs_inactive would fulfill the misleading adjective.

**Review by skirvan44 — COMMENTED** [review]
From a usability/understandability perspective I think running is preferable to inactive.  Its a usability BKM for things to be consistently logically positive.  i.e. "active" is better than "not active" as a field.

Also please note that there are a lot of rebuild tests that depend on this field so make sure they are all updated to reflect any change.

**Comment by wangshilong** [comment]
> I don't think this is quite what the ticket wanted to do. We can't just rename the field because rs_done != rs_running. To change it to rs_running, we would need to change the implementation to ensure that it's set if rebuild is running. In the spirit of the ticket, just renaming rs_done to rs_inactive would fix the stated concern that rs_done implies it did something whereas rs_inactive just means it's not running. To determine if it has run, one would need to check the status of the pool to see if there are DOWNOUT targets though that may be insufficient.

Try another direction PR:
https://github.com/daos-stack/daos/pull/7885

> 
> It may be worth asking whether the intention of the ticket was a feature request to provide a "running" flag or just pointing out that "done" is misleading or perhaps both. Anyway, if you are going to fix the meaning of the flag, I think just replacing rs_running in your patch with rs_inactive would fulfill the misleading adjective.

**Pull request closed without merging**